spec.md

```diff
@@ -15,4 +14,569 @@ The key words "unspecified", "undefined", and "implementation-defined" are to be
 An implementation is not compliant if it fails to satisfy one or more of the MUST, REQUIRED, or SHALL requirements for the protocols it implements.
 An implementation is compliant if it satisfies all the MUST, REQUIRED, and SHALL requirements for the protocols it implements.
 
+## Terminology
+
+| Term              | Definition                                       |
+|-------------------|--------------------------------------------------|
+| Bucket            | A flat organization of container that store files.                          |
+| COSI  System              | Container Orchestration system/ Container Object Storage Implementation, communicates with Plugins using COSI service RPCs.                                     |
+| SP                | Storage Provider, the vendor of a COSI plugin implementation.                                                          |
+| RPC               | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).                                         |
+| Node              | A host where the user workload will be running, uniquely identifiable from the perspective of a Plugin by a node ID. |
+| Plugin            | Aka “plugin implementation”, a gRPC endpoint that implements the COSI Services.                                        |
+| Plugin Supervisor | Process that governs the lifecycle of a Plugin, MAY be the COSI.                                                        |
+| Workload          | The atomic unit of "work" scheduled by a COSI System. This MAY be a container or a collection of containers.                   |
+
+## Objective
+
+To define a standard “Container Object Storage Interface” (COSI) that will enable storage vendors (SP) to develop a plugin once and have it work across multiple Kubernetes Clusters and possibly on other CO.
+
+### Goals in MVP
+
+The Container Object Storage Interface (COSI) will
+
+* Enable SP authors to write one COSI compliant Plugin that “just works” across all Kubernetes based clusters that implement COSI.
+* Define API (RPCs) that enable:
+  * Dynamic provisioning and de-provisioning of a bucket.
+  * Ability to provide access control mechanism to read write to buckets
```

Comment:
`  * Ability to define bucket access control mechanisms`

Reply:
Done